### PR TITLE
LG: add context to parsing error

### PR DESCRIPTION
**Pull request by Danieladu**
for Antlr error, we use such error format: 
```
[Error]:
my template body(params. invalid)
                       ^^
invalid parameters format
```

### CODE REVIEW

**Comment by tomlm** [comment]
can you run these unit tests again?

**Comment by Danieladu** [comment]
sure

> can you run these unit tests again?

sure

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/ErrorListener.cs [hunk 1]
```diff
@@ -9,7 +10,19 @@ public class ErrorListener : BaseErrorListener
     {
         public override void SyntaxError([NotNull] Antlr4.Runtime.IRecognizer recognizer, [Nullable] IToken offendingSymbol, int line, int charPositionInLine, [NotNull] string msg, [Nullable] RecognitionException e)
         {
-            throw new Exception($"[ERROR]: syntax error at line {line}:{charPositionInLine} {msg}");
+            var lineStrArray = offendingSymbol.InputStream.GetText(new Interval(0, int.MaxValue)).Split('\n');
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/ErrorListener.cs]: you should still output this line number and char number, i expect this error message will be

[Error] syntax error at line 4, char 5-10
line3:   # templateName
line4:    - [t2]
               ^^ no such template
line5：# another template

**Comment by tomlm** [comment]
could you requeue the build?

**Comment by Danieladu** [comment]
close it for we will have a new error handler and highlight function, but not in this branch

**Pull request closed without merging**